lib/fizzy/execute.cpp

```diff
@@ -254,41 +254,29 @@ inline constexpr T rotr(T lhs, T rhs) noexcept
     return (lhs >> k) | (lhs << (num_bits - k));
 }
 
-inline uint32_t clz32(uint32_t value) noexcept
+constexpr uint32_t clz32(uint32_t value) noexcept
```

Comment:
We can replace these with a template and use them as `clz<uint32_t>` below. Would cat this code duplication into half.

Reply:
Maybe we dont even need this wrapper layer, the only thing it does is explicit casting of the return value.